src/Lykke.Service.BlockchainWallets.Services/BlockchainIntegrationService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Common.Log;
+using Lykke.Service.BlockchainApi.Client;
+using Lykke.Service.BlockchainSignService.Client;
+using Lykke.Service.BlockchainWallets.Core.Services;
+using Lykke.Service.BlockchainWallets.Core.Settings.BlockchainIntegrationSettings;
+
+
+namespace Lykke.Service.BlockchainWallets.Services
+{
+    public class BlockchainIntegrationService : IBlockchainIntegrationService
+    {
+        private readonly ImmutableDictionary<string, BlockchainApiClient>         _apiClients;
+        private readonly ImmutableDictionary<string, BlockchainSignServiceClient> _signServiceClients;
+
+
+        public BlockchainIntegrationService(
+            BlockchainsIntegrationSettings settings,
+            ILog log)
+        {
+            _apiClients = settings.Blockchains.ToImmutableDictionary
+            (
+                x => x.Type,
+                y => new BlockchainApiClient(log, y.ApiUrl)
+            );
+
+            _signServiceClients = settings.Blockchains.ToImmutableDictionary
+            (
+                x => x.Type,
+                y => new BlockchainSignServiceClient(y.SignFacadeUrl, log)
+            );
+        }
```

Comment:
IReadOnlyDictionary will be better in both cases, since your neve alter these dicts. But Immutable is ok too

Reply:
Just a habit. Immutables protect you from unexpected side affects in multi-thread environments.

Reply:
Agree, but IReadOnlyDictionary too.